Lib/fontTools/pens/hashPointPen.py

```diff
@@ -0,0 +1,46 @@
+# Modified from https://github.com/adobe-type-tools/psautohint/blob/08b346865710ed3c172f1eb581d6ef243b203f99/python/psautohint/ufoFont.py#L800-L838
+import hashlib
+from fontTools.pens.pointPen import AbstractPointPen
+from fontTools.pens.transformPen import TransformPointPen
+
+
+class HashPointPen(AbstractPointPen):
+    DEFAULT_TRANSFORM = (1, 0, 0, 1, 0, 0)
+
+    def __init__(self, glyph, glyphSet=None):
+        self.glyphset = glyphSet
+        self.width = round(getattr(glyph, "width", 1000), 9)
+        self.data = ["w%s" % self.width]
+
+    def getHash(self):
+        data = "".join(self.data)
+        if len(data) >= 128:
+            data = hashlib.sha512(data.encode("ascii")).hexdigest()
+        return data
+
+    def beginPath(self, identifier=None, **kwargs):
+        pass
+
+    def endPath(self):
+        pass
+
+    def addPoint(
+        self,
+        pt,
+        segmentType=None,
+        smooth=False,
+        name=None,
+        identifier=None,
+        **kwargs
+    ):
+        if segmentType is None:
+            pt_type = ""
+        else:
+            pt_type = segmentType[0]
+        self.data.append("%s%s%s" % (pt_type, repr(pt[0]), repr(pt[1])))
```

Comment:
Perhaps use an f-string here?
```python
        self.data.append(f"{pt_type}{pt[0]}{pt[1]}")
```
Btw. this is not robust against a change of, say, `(1, 11)` to `(11, 1)`, as both will be represented as `111`.

Reply:
Ha, nice catch!